File: c/darts/src/darts.c
#include <math.h>
#include <stdint.h>
#include "darts.h"

uint8_t score(coordinate_t pos) {

        float radius = sqrt(pos.x * pos.x + pos.y * pos.y);
        int score = 0;

        if (radius > 10) {
            // Outside the circle, no score
        } else if (radius > 5) {
            // Outer radius
            score = 1;
        } else if (radius > 1) {
            // Middle radius
            score = 5;
        } else if (radius >= 0) {
            // Inner radius
            score = 10;
        }

        return score;
}

File: go/darts/darts.go
package darts

import "math"

const (
	noScore     = 0
	outerScore  = 1
	middleScore = 5
	innerScore  = 10
)

func Score(x, y float64) int {

	radius := math.Sqrt(x*x + y*y)
	var score int

	switch {
	case radius > 10:
		score = noScore
	case radius > 5:
		score = outerScore
	case radius > 1:
		score = middleScore
	case radius >= 0:
		score = innerScore
	default:
		score = noScore
	}

	return score
}
